### PR TITLE
Move workload related classes from dispatcher to respective workloads

### PR DESCRIPTION
**Pull request by manju956**
- Move EchoResult and HeartDiseaseEvalFactory classes from
  dispatcher to respective workloads. This decouples dispacther code from workloads

Signed-off-by: manju956 <manjunath.a.c@intel.com>

### CODE REVIEW

File: examples/apps/CMakeLists.txt [hunk 1]
```diff
@@ -15,6 +15,8 @@
 CMAKE_MINIMUM_REQUIRED(VERSION 3.2 FATAL_ERROR)
 
 INCLUDE(CMakeVariables.txt)
+SET(GENERIC_PRIVATE_INCLUDE_DIRS "." "${TCF_TOP_DIR}/tc/sgx/common"
+	"${TCF_TOP_DIR}/tc/sgx/common/work_order_interface")
```
Comment (manojgop) [inline examples/apps/CMakeLists.txt]: What's the dependency with ${TCF_TOP_DIR}/tc/sgx/common"

File: examples/apps/echo/workload/echo_factory.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
+
```
Comment (manojgop) [inline examples/apps/echo/workload/echo_factory.cpp]: No need of echo_factor.h and echo_factory.cpp. These could be moved to echo.h. EchoResultImpl  in echo.h is not required. Process in EchoResultImpl  can be a private function in EchoResult. 
Similarly no need of heart_disease_evaluation_factory.cpp

File: common/sgx_workload/workload_dispatcher.cpp [hunk 3]
```diff
@@ -16,96 +16,18 @@
 #include <string>
 #include "work_order_data.h"
 #include "workload_processor.h"
-#include "echo/workload/echo.h"
-#include "heart_disease_eval/workload/heart_disease_evaluation.h"
+#include "echo/workload/echo_factory.h"
+#include "heart_disease_eval/workload/heart_disease_evaluation_factory.h"
```
[the remark this answers is not in the document]
Comment (manju956) [reply common/sgx_workload/workload_dispatcher.cpp]: Ok

File: examples/apps/CMakeLists.txt (hunk 1 above)
Comment (manju956) [reply examples/apps/CMakeLists.txt]: work_order_interface includes error.h from ${TCF_TOP_DIR}/tc/sgx/common.

File: examples/apps/echo/workload/echo_factory.cpp (hunk 2 above)
Comment (manju956) [reply examples/apps/echo/workload/echo_factory.cpp]: Ok

**Pull request merged**